srtcore/utilities.h

```diff
@@ -440,6 +440,70 @@ struct DynamicStruct
 };
 
 
+/// Fixed-size array template class.
+namespace srt {
+
+template <class T>
+class FixedArray
+{
```

Comment:
1. utilities.h is included by common.h
2. common.h use something defined by utilities.h, e.g., HAVE_FULL_CXX11
3. Something defined by common.h can be useful for utilities.h, e.g., SRT_ASSERT, SRT_STATIC_ASSERT, but can't due to circular header include

utilities.h is more than utility, it needs some cleanup work. Before that, I prefer the "include what you use" method, and use a separate header file. Of course, the cleanup work can be done after the patch. So the comments above doesn't related to the patch strictly.